mlir/lib/Dialect/Linalg/IR/LinalgInterfaces.cpp

```diff
@@ -1172,19 +1240,26 @@ LogicalResult mlir::linalg::detail::verifyStructuredOpInterface(Operation *op) {
 
     // Domain must be consistent.
     unsigned numLoops = linalgOp.getNumLoops();
-    if (indexingMap.getNumDims() != numLoops)
+    if (!isUserDefinedSemantic && indexingMap.getNumDims() != numLoops)
```

Comment:
You can drop the `isUserDefinedSemantic` here. This should hold irrespective of "user defined semantic".

Also `isUserDefinedSemantic` is a bit misleading. Maybe `hasNonDefaultSemantics` is better.

Reply:
Dropped.